### PR TITLE
Visitor can see different products on a restaurants menu

### PR DESCRIPTION
**Pull request by yasmineezequiel**
[PT-link](https://www.pivotaltracker.com/story/show/168829429)
Yasmine & Pia

# PR contains
In this PR we've created a menu index-page where we display different products from a restaurants menu. 

## This PR contains
- Feature-test
  - with DB-table for pre-existing products
- Menu controller
- Menu index page
- Product model 
  - with name, description and price
- FactoryBot
- Product spec tests
- List with content on menu index page

## What we've learned
- Learned how to convert code in .erb into .haml
- Deeper understanding of working with BDD. 

Screenshot 2019-10-01 at 10 43 45

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+get 'menu/index'
+
+root 'menu#index'
```
Comment (faraznaeem) [inline config/routes.rb]: Please use the proper syntax when writing routes, and make sure that you use resourceful routing.

**Review by faraznaeem — CHANGES_REQUESTED** [review]
This feature seems incomplete, the story states that we should see products on the resturants menu. However, this feature only has hardcoded information on the landing page. Please add the Product model and review this feature again to meet the requirements of the story.

**Review by clalil — APPROVED** [review]

**Review by clalil — CHANGES_REQUESTED** [review]
- Remove all br
- Change hash in basic_steps to product.

**Review by clalil — APPROVED** [review]

**Review by DIzosimov — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]
Looks good!

**Review by piavW — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

File: app/models/product.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+  validates :name, presence: true
+  validates :description, presence: true
+  validates :price, presence: true
```
Comment (faraznaeem) [inline app/models/product.rb]: Refactor the validations

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get 'menu/index'
```
Comment (faraznaeem) [inline config/routes.rb]: as we say in swedish: aja baja

File: features/support/menu.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      | vesuvio       | cheese, tomato sauce, ham       | 70    |
+      | calzone       | cheese, tomato sauce, ham       | 80    |
+      | ice-cream     | strawberry                      | 40    |
+      | ice-cream     | vanilla                         | 40    |
+      | ice-cream     | chocolate                       | 40    |     
+  
+  Scenario: Be able to see menu
+    Given I visit the page 
+    Then I should see 'Menu'
+    And I should see "garlic bread"
```
Comment (faraznaeem) [inline features/support/menu.feature]: You should test all of the products and their attributes in the background and not only garlic bread

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: features/basic_steps.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Given("I visit the page") do
+  visit root_path
+end
+
+Then("I should see {string}") do |string|
```
Comment (oliverochman) [inline features/basic_steps.rb]: this is an assertion, please put it in an assertion_steps file

File: features/support/menu.feature [hunk 6]
```diff
@@ -0,0 +1,25 @@
+Feature: Visitor can view menu
+As a visitor
```
Comment (oliverochman) [inline features/support/menu.feature]: line 2 to 4 should be indented

File: db/seeds.rb [hunk 7]
```diff
@@ -1,7 +1,9 @@
 # This file should contain all the record creation needed to seed the database with its default values.
 # The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
 #
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+product = Product.create(name: "Garlic Bread", description: "garlic, bread, butter", price: 30)
```
Comment (oliverochman) [inline db/seeds.rb]: products = [
  Product.create(name: "Garlic Bread", description: "garlic, bread, butter", price: 30) ,
  Product.create(name: "Margherita", description: "cheese, tomato sauce, basil", price: 70),
  Product.create(name: "Vesuvio", description: "cheese, tomato sauce, ham", price: 70),
  Product.create(name: "Calzone", description: "cheese, tomato sauce, ham", price: 80),
  Product.create(name: "Ice-cream", description: "vanilla", price: 40),
  Product.create(name: "Ice-cream", description: "strawberry", price: 40)
]

**Review by oliverochman — APPROVED** [review]
lgtm!

**Pull request merged**